server/models/users.py

```diff
@@ -1,13 +1,67 @@
-from app import db
+from app import create_app, db
+from passlib.hash import pbkdf2_sha256 as sha256
+from marshmallow import Schema, fields, ValidationError, pre_load
 
 class UserModel(db.Model):
     __tablename__ = 'users'
 
-    # Should autoincrement (this is the implicit default with PK set to True; result is a bigserial column.
+    # Assign database fields
+    # Autoincrement is implicit default with PK set to True
     id = db.Column(db.BigInteger, primary_key=True)
-    username = db.Column(db.String(120), unique=True, nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
     password = db.Column(db.String(120), nullable=False)
 
     def save_to_db(self):
         db.session.add(self)
         db.session.commit()
+
+    # return a user’s data if there is match by email
+    @classmethod
+    def find_by_email(cls, email):
+        return cls.query.filter_by(email=email).first()
+
+    @classmethod
+    def return_all(cls):
+        def to_json(x):
+            return {
+                'email': x.email,
+                'password': x.password
+            }
+        return {'users': list(map(lambda x: to_json(x), UserModel.query.all()))}
+
+    @classmethod
+    def delete_all(cls):
+        try:
+            num_rows_deleted = db.session.query(cls).delete()
+            db.session.commit()
+            return {'message': '{} rows deleted'.format(num_rows_deleted)}
+        except:
+            return {'message': 'Something went wrong'}
+
+    @staticmethod
+    def generate_hash(password):
+        return sha256.hash(password)
```

Comment:
sha256 is not a great password hashing function (https://stackoverflow.com/questions/11624372/best-practice-for-hashing-passwords-sha256-or-sha512)... we are also not using a salt

Can we use bcrypt and flask_bcrypt -  https://flask-bcrypt.readthedocs.io/en/latest/

Reply:
noted - working on this now